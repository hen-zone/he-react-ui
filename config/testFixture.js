import createTestContext from 'react-cosmos-test/enzyme';

export default function testFixture(fixture, variant) {
  const { mount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component;
  mount();
  test(`<${displayName ||
    name} /> rendered correctly with ${variant} fixture`, () =>
    expect(getWrapper()).toMatchSnapshot());
}
