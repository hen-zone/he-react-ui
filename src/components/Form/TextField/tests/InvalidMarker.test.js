import fixture from '../fixtures/InvalidMarker.fixture.js'
import testFixture from '../../../../../config/testFixture'

testFixture(fixture, 'InvalidMarker')
