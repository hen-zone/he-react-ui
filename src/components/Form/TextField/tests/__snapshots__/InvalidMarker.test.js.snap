// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> rendered correctly with InvalidMarker fixture 1`] = `
<TextField
  disabled={false}
  error="Something isn't right"
  id="demo"
  inline={false}
  isValid={true}
  label="First Name"
  marker={true}
  name="demo"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  value="Bruce"
>
  <div
    className="outer hasMarker"
  >
    <div
      className="block"
    >
      <input
        className="input"
        disabled={false}
        id="demo"
        name="demo"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        value="Bruce"
      />
      <Tick
        className="marker"
        fill="currentColor"
        height={32}
        width={32}
      >
        <svg
          className="marker"
          fill="currentColor"
          height={32}
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="M17.207 7.707a1 1 0 0 1 1.415 1.414l-7.998 8.003a1 1 0 0 1-1.414 0l-4.003-4.003a1 1 0 1 1 1.414-1.414l3.296 3.296 7.29-7.296z"
          />
        </svg>
      </Tick>
      <label
        className="label floating"
        htmlFor="demo"
      >
        First Name
      </label>
    </div>
    <label
      className="description"
      htmlFor="demo"
    >
      Something isn't right
    </label>
  </div>
</TextField>
`;
